### PR TITLE
PR to add viewModels to UI 

### PR DESCRIPTION
**Pull request by Santboneless**
This PR has for goal to seperate the UI and create viewModels for them
First class done for now is the friendsList but I will add for every screen I can before the end of the project

### CODE REVIEW

**Comment by Santboneless** [comment]
Done with the splitting into viewModels and UI
I didn't do it for maps yet and some other classes that are not finished yet so I will add them later on
I need help to manually test every feature and see if there is crashes or unseen bug on emulator

**Comment by Liototo** [comment]
I couldn't analyse the code in detail since it's a huge PR, but it all seems cleaner and more modular, and manually testing the app on my phone everything seems to work as well as before! I'll approve your changes once the CI passes and you mark the PR as ready for review!

**Review by Liototo — APPROVED** [review]
It's all good now, well done!

**Pull request merged**
